### PR TITLE
Make more static arrays const

### PR DESCRIPTION
**Pull request by ccawley2011**
This should help reduce memory usage a bit on platforms with limited writable memory.

This also fixes most of the `-Wcast-qual` warnings to ensure that nothing is likely to write to static memory. One warning remains with `execvp` in the depackers, but there's probably not much that can be done about that at the moment.

### CODE REVIEW

**Comment by AliceLR** [comment]
Initial thoughts, will do a better review later:

* Since `HIO_HANDLE`s currently are read-only, it might be possible/beneficial to make `const` the *only* option for opening buffers rather than to add a new function right now. It looks like the helpers that hio.c uses internally already take `const` pointers.
* The lhasa changes should be offered upstream etc.

**Comment by sezero** [comment]
>   Since `HIO_HANDLE`s currently are read-only, it might be possible/beneficial to make `const` the _only_ option for opening buffers rather than to add a new function right now. It looks like the helpers that hio.c uses internally already take `const` pointers.

Agreed

>   The lhasa changes should be offered upstream etc.

Our lhasa already has const changes, don't know how doable it is to offer the existing ones + the ones from this patchset upstream..

**Comment by ccawley2011** [comment]
> Since HIO_HANDLEs currently are read-only, it might be possible/beneficial to make const the only option for opening buffers rather than to add a new function right now. It looks like the helpers that hio.c uses internally already take const pointers.

I've adapted this a bit, but the reason for this is to avoid casting away the const qualifier in order to pass it to `free` for buffers allocated by the depackers.

> The lhasa changes should be offered upstream etc.

Done:
* nothings/stb#1660
* fragglet/lhasa#53

The lhasa changes involve changing the public API, though.

File: src/hio.c [hunk 1]
```diff
@@ -499,7 +499,7 @@ int hio_reopen_mem(const void *ptr, long size, int free_after_use, HIO_HANDLE *h
 
 	ret = hio_close_internal(h);
 	if (ret < 0) {
-		m->free_after_use = 0;
+		m->ptr_free = 0;
```
Comment (sezero) [inline src/hio.c:502]: NULL

**Comment by AliceLR** [comment]
> I've adapted this a bit, but the reason for this is to avoid casting away the const qualifier in order to pass it to `free` for buffers allocated by the depackers.

Oh yes, you're completely correct, sorry.

> Done:
> 
> * [Make more static arrays in stb_vorbis const nothings/stb#1660](https://github.com/nothings/stb/pull/1660)
> * [Make more static arrays const fragglet/lhasa#53](https://github.com/fragglet/lhasa/pull/53)
> 
> The lhasa changes involve changing the public API, though.

Thanks! Will properly review tonight...

**Pull request merged**